packages/x-charts/src/BarChart/seriesConfig/seriesProcessor.ts

```diff
@@ -61,6 +60,8 @@ const seriesProcessor: SeriesProcessor<'bar'> = (params, dataset) => {
       completedSeries[id] = {
         layout: 'vertical',
         labelMarkType: 'square',
+        minBarSize: 0,
+        valueFormatter: series[id].valueFormatter ?? ((v) => (v == null ? '' : v.toLocaleString())),
```

Comment:
Could we move the default value formatter outside so that it's a stable function? Should help reduce re-renders.